google/resource_attached_disk.go

```diff
@@ -0,0 +1,286 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	compute "google.golang.org/api/compute/v1"
+)
+
+func resourceAttachedDisk() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAttachedDiskCreate,
+		Read:   resourceAttachedDiskRead,
+		Update: resourceAttachedDiskUpdate, // TODO remove this method? make sure it's not called? etc
+		Delete: resourceAttachedDiskDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAttachedDiskImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(300 * time.Second),
+			Update: schema.DefaultTimeout(240 * time.Second),
+			Delete: schema.DefaultTimeout(240 * time.Second),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"attached_disk": {
+				Type:     schema.TypeString,
+				Required: true,
+				// TODO (chrisst) - add force new here so that people have to recreate instead of update
+				// TODO (chrisst) change the name of this attribute
+			},
+			"attached_instance": {
+				Type:     schema.TypeString,
+				Required: true,
+				// TODO (chrisst) - add force new here so that people have to recreate instead of update
+				// TODO (chrisst) change the name of this attribute
+			},
+			"project": {
+				// Adding project as computed so that import can use parseImportId
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"zone": {
+				// Adding zone as computed so that import can use parseImportId
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				// TODO play with this to determine if somebody can change between PERSISTANT vs SCRATCH.
+				// And see if it's worth allowing a user to pass in
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"self_link": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAttachedDiskCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	instance := d.Get("attached_instance").(string)
+	zone, err := getZoneForAttachedDisk(d, config, instance)
+	if err != nil {
+		return err
+	}
+	instanceName := GetResourceNameFromSelfLink(instance)
+	diskName := GetResourceNameFromSelfLink(d.Get("attached_disk").(string))
+
+	attachedDisk := compute.AttachedDisk{}
```

Comment:
nit: it's a bit cleaner to do
```
attachedDisk := compute.AttachedDisk{
  DeviceName: [...],
  Source: [...],
}
```
we also tend to use the pointer form here (`&compute.AttachedDisk`), but I don't think there's a real reason for it besides just consistency